File: polimorfismo/garrafa.cs
public class Garrafa
{
    public virtual void finalidade()
    {
        Console.WriteLine("Garrafa genêrica");
    }

}

public class GarrafaTermica : Garrafa
{
    public virtual void finalidade()
    {
        Console.WriteLine("Manter temperatura");
    }

}

public class GarrafaVidro : Garrafa
{
    public virtual void finalidade()
    {
        Console.WriteLine("Manter o sabor");
    }

}

public class Barril : Garrafa
{
    public virtual void finalidade()
    {
        Console.WriteLine("armazenar grandes quantidades");
    }

}

File: polimorfismo/garrafa.dart
class Garrafa{
    void finalidade(){
        print("Garrafa genêrica");
    }  
}

class GarrafaTermica extends Garrafa{
    @override
    void finalidade(){
        print("Manter a temperatura");
    }
}

class GarrafaVidro extends Garrafa{
    @override
    void finalidade(){
        print("Manter o sabor");
    }
}

class Barril extends Garrafa{
    @override
    void finalidade(){
        print("armazenar grandes quantidades");
    }
}
